Add about us branding page

mnajdova opened the pull request:
Still WIP, but you can check the progress here - https://deploy-preview-24095--material-ui.netlify.app/branding/about/. 

We are implementing the first page of the rebranding. This will take some time, there are a lot of content to cover. 

TODO:
- [x] Replace icons with correct svgs & add missing svgs
- [x] Implement responsiveness 
- [ ] Update links & buttons actions
- [x] Review text & do necessary changes
- [x] Review the spacings between elements
- [x] Update texts with identity fn for easier replacement with `t()` {translation} (only in the common components for now)

mnajdova commented:
@mbrookes @oliviertassinari I think we can start with first round of review around this one. I left comments where I wasn't sure about which links we should provide, and we will need to make the input fields in the header and footer actionable, they are not doing anything at this moment.

oliviertassinari commented on docs/src/modules/components/branding/UnderlinedText.js:
```diff
@@ -0,0 +1,8 @@
+
+const UnderlinedText = styled('span')({
+  textDecoration: 'underline',
+  textDecorationColor: '#00C8FF',
```
Also:

- there is an opacity of 40%
- the thickness is 10px. I have found https://css-tricks.com/styling-underlines-web/ on this problem

mnajdova replied:
Fixed by uising `backgroundImage`. Looks much better now:

![image](https://user-images.githubusercontent.com/4512430/103131381-d4924700-46a0-11eb-9354-9f9366d715a9.png)

oliviertassinari replied:
Perfect! 👌

oliviertassinari commented on docs/src/modules/components/branding/SearchAppBar.js:
```diff
@@ -0,0 +1,135 @@
+          </IconButton>
+        </Toolbar>
+      </AppBar>
+      <Drawer anchor={'right'} open={open} onClose={toggleDrawer(false)}>
```
Should we care about the difference? Historically, we have been using `"` anything we could, e.g. class names. But, I have also seen Damien use this approach, so I wonder

```suggestion
      <Drawer anchor="right" open={open} onClose={toggleDrawer(false)}>
```

mnajdova replied:
Not sure, I'd except prettier to fix it if we care about it :upside_down_face: I can replace those, but I can't guarantee I won't add them again 😄

oliviertassinari commented on docs/src/modules/components/branding/Quote.js:
```diff
@@ -0,0 +1,43 @@
+    </Grid>
+  );
+})({
+  background: '#F3F6F9',
```
I think that we can start with the names used in the Sketch file and later rename them to something we prefer.

```suggestion
  backgroundColor: theme.palette.greyF3,
```

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,90 @@
+  typography: {
+    h1: {
+      fontSize: 72,
+      fontWeight: 700,
```
```suggestion
      fontWeight: 700,
      letterSpacing: '-2px',
      lineHeight: 1.111,
```

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,90 @@
+    },
+    h2: {
+      fontSize: 52,
+      fontWeight: 700,
```
```suggestion
      fontWeight: 700,
      letterSpacing: '-1.5px',
      lineHeight: 1.154,
```

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,90 @@
+    },
+    h4: {
+      fontSize: 24,
+      fontWeight: 700,
```
```suggestion
      fontWeight: 700,
      letterSpacing: '-0.5px',
      lineHeight: 1.25,
```

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,90 @@
+    },
+    h3: {
+      fontSize: 36,
+      fontWeight: 700,
```
```suggestion
      fontWeight: 700,
      
      1.222
```

mnajdova commented on docs/src/modules/components/branding/icons/Search.js:
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import SvgIcon from '@material-ui/core/SvgIcon';
+
+const Search = React.forwardRef((props, ref) => (
```
I tried to use this, but I thought it's internal, that's why I decided to go with the `SvgIcon`. Is that not the case?

oliviertassinari replied:
It's a public helper: https://next.material-ui.com/components/icons/#createsvgicon

mnajdova replied:
Great, will update the icons 👍

oliviertassinari commented on docs/src/modules/components/branding/Footer.js:
```diff
@@ -0,0 +1,332 @@
+      display: 'none',
+    },
+  },
+
```
Shouldn't only add a blank line when usin  CSS template strings?

```suggestion
```

oliviertassinari commented on docs/src/modules/components/branding/Footer.js:
```diff
@@ -0,0 +1,332 @@
+  },
+
+  '& h2': {
+    fontWeight: 700,
```
```suggestion
    fontWeight: ontWeight: theme.typography.fontWeightBold,
```

oliviertassinari commented on docs/src/modules/components/branding/Footer.js:
```diff
@@ -0,0 +1,332 @@
+import LinkedInIcon from '@material-ui/icons/LinkedIn';
+import { useUserLanguage, useTranslate } from 'docs/src/modules/utils/i18n';
+import { experimentalStyled as styled } from '@material-ui/core/styles';
+import Logo from './Logo';
```
I think that we should use absolute imports so we can move code around
```suggestion
import Logo from 'docs/src/modules/components/branding/Logo';
```

x the other places

oliviertassinari reviewed (commented):
I will continue later on. Great progress :D.


Other things I haven noticed:

- The background colored sections are supposed to bleed. We can't have a single `<Container>` component. We need to have one per block, so we can set the background color. For instance: 
<img width="1333" alt="Capture d’écran 2020-12-24 à 17 51 48" src="https://user-images.githubusercontent.com/3165635/103099947-baa92380-4610-11eb-80bb-17e790df6fea.png">

- For the button. We don't have guidelines for the interactions. How about we remove the ripple, and have a hover/focus/active behavior as bootstrap?
- I have noticed a couple of issues using https://wave.webaim.org/. It might be overkill to solve all the issues but caring about the most important ones could have a good ROI.

mnajdova commented on docs/src/modules/components/branding/AppSearch.js:
```diff
@@ -0,0 +1,171 @@
+
+  React.useEffect(() => {
+    macOS = window.navigator.platform.toUpperCase().indexOf('MAC') >= 0;
+  }, []);
```
I've removed the effect but had to add a check if `window` is not undefined.

oliviertassinari replied:
Yeah, well, now, it jumps when reloading 😁 

![reload](https://user-images.githubusercontent.com/3165635/103140848-f82eaf00-46eb-11eb-8476-884ebc9312d5.gif)

I think that we need some way to share this logic: https://github.com/mui-org/material-ui/blob/186234662da4ae330d5cb882096ab5428249cf7a/docs/src/modules/components/AppFrame.js#L48

mnajdova replied:
I've added `React.lazy` here too, hopefully, it is better now 😅

mnajdova replied:
Adding `React.lazy` is breaking the build - https://app.netlify.com/sites/material-ui/deploys/5fea0b930bf8d2000848112f
https://reactjs.org/docs/error-decoder.html/?invariant=295 :\

mnajdova commented:
> The background colored sections are supposed to bleed horizontally. We can't have a single <Container> component. We need to have one per block, so we can set the background color. For instance:

I don't quite understand the issue here. Can you maybe add a picture of the difference in look. (it's probably just my eyes not seeing the difference, or ignoring it :))

> For the button. We don't have guidelines for the interactions. How about we remove the ripple, and have a hover/focus/active behavior as default Bootstrap?

Removed the ripple, added hover background for the type of buttons we have and added outline for focus. Currently the outline is `vividBlue` for all Buttons, but we can change this easily. Not sure if it's best to keep the same color as the background and just change shade or to have some double outline (black and white) to go well on all backgrounds (fluent had something like this)

> I have noticed a couple of issues using https://wave.webaim.org/. It might be overkill to solve all the issues but caring about the most important ones could have a good ROI.

I fixed most of the issues around missing `alt` we can do a final review before merging again.

oliviertassinari commented on docs/src/modules/components/branding/Logo.js:
```diff
@@ -0,0 +1,27 @@
+  );
+}
+
+Logo.propTypes = {
+  href: PropTypes.string,
+};
```
What do you think about writing these new pages in TypeScript?

```suggestion
```

mnajdova replied:
Makes sense to move the new pages to `Typescript`. I'll work on this tomorrow 👍

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,151 @@
+    background: {
+      default: '#FFF',
+    },
+    vividBlue: '#00C8FF',
+    greyF3: '#F3F6F9',
+    greyEAE: '#EAEEF3',
+    grey5A: '#5A6978',
+    azure: '#007FFF',
```
I think that we can all the color upfront, they should all be needed later down the road:

```suggestion
    vividBlue: '#00C8FF',
    emeral: '#21CC66',
    sizzlingRed: '#FF505F',
    sunglow: '#FFC846',
    
```

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,151 @@
+              left: 0,
+              right: 0,
+              borderRadius: '2px',
+              border: '2px solid #00C8FF', // vivid blue outline when focused
```
Haha, the classic. See #21757.

mnajdova replied:
Yeah, I knew we had an issue where we discussed this. I will at least for this moment extract these in constants to avoid using hardcoded values..

oliviertassinari replied:
What do you think about going with the latest conclusion of this issue. So far, it seemed that the strategy was to use and document https://github.com/mui-org/material-ui/issues/21757#issuecomment-657226028.

mnajdova replied:
Sounds good, will refactor it 👍

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,151 @@
+      styleOverrides: {
+        root: {
+          '&.Mui-focusVisible': {
+            ':after': {
```
Did you consider box-shadow?

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,151 @@
+      },
+      styleOverrides: {
+        root: {
+          '&.Mui-focusVisible': {
```
Stripe and Bootstrap are using `:focus` over `:focus-visible` but that' probably a mistake, I guess we are good.

https://user-images.githubusercontent.com/3165635/103140312-4a200680-46e5-11eb-8eff-d444a20b8301.mp4

mnajdova replied:
Added `"&:active"` that calculates a different background color.

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,151 @@
+  components: {
+    MuiButton: {
+      defaultProps: {
+        disableRipple: true,
```
https://next.material-ui.com/components/buttons/#contained-buttons

```suggestion
        disableRipple: true,
        disableElevation: true,
```

oliviertassinari commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,151 @@
+          props: { variant: 'contained', color: 'secondary' },
+          style: {
+            '&:hover': {
+              backgroundColor: '#19344f',
```
How did you find this value? Should it be computed with our color utils from the baseline color?

mnajdova replied:
I hand picked the value. However, I refactor it to use the `lighten` color util.

oliviertassinari commented on docs/pages/branding/about.js:
```diff
@@ -0,0 +1,1067 @@
+import NavigateNextIcon from '@material-ui/icons/NavigateNext';
+import { experimentalStyled as styled } from '@material-ui/core/styles';
+import QuestionAnswerIcon from '@material-ui/icons/QuestionAnswer';
+
```
```suggestion
```

oliviertassinari commented on docs/pages/branding/about.js:
```diff
@@ -0,0 +1,1067 @@
+            <Grid item lg={6} xs={12} className="MuiGrid-relative">
+              <Box
+                component="img"
+                src="/static/images/branding/AboutUs-1.jpg"
```
I think that:

- we should isolate the images per page, to make them easier to delete in the future, once we revamp the page, maybe in 2 years.
- it's best to stick to a single filename convention, so far, kebab-case.
- we can flatten the import structure.

```suggestion
                src="/static/about/top-left.jpg"
```

mnajdova replied:
Done

oliviertassinari reviewed (commented):
> I don't quite understand the issue here. Can you maybe add a picture of the difference in look. (it's probably just my eyes not seeing the difference, or ignoring it :))

Sure, I see 4 issues with the current approach:

- The background image is distorted

<img width="234" alt="Capture d’écran 2020-12-25 à 19 38 48" src="https://user-images.githubusercontent.com/3165635/103140606-e4357e00-46e8-11eb-9ce5-5109bb5fd4a9.png">

- The dark blue background should take the whole vertical space

<img width="1354" alt="background" src="https://user-images.githubusercontent.com/3165635/103140652-7178d280-46e9-11eb-9f22-3a99833cd71d.png">

- The container is [fixed](https://material-ui.com/components/container/#fixed), not [fluid](https://material-ui.com/components/container/#fluid)
- The page isn't using the `<Container>` component.

mnajdova commented on docs/src/modules/components/branding/Page.js:
```diff
@@ -0,0 +1,90 @@
+    },
+  },
+  typography: {
+    h1: {
```
Adding these fonts, make the regular <p /> texts looks very much bolded

![image](https://user-images.githubusercontent.com/4512430/103215089-a9af2980-4912-11eb-9f4e-158002590a1a.png)

Can you check how does it look on your machine?

oliviertassinari replied:
It looks fine on my machine, but I have installed Inter manually. Did we update?

https://github.com/mui-org/material-ui/blob/506181fde08588ee7441a09800e3ed9a218bcba2/docs/pages/_app.js#L304-L311

mnajdova replied:
Updated the fonts array and it looks good locally now too. 👍 I used the same URL as the `Roboto`, just changes the name to `Inter`

mnajdova commented:
> Sure, I see 4 issues with the current approach:
> 
> 1. The background image is distorted
> 2. The dark blue background should take the whole horizontal space
> 3. The container is fixed, not fluid as most websites nowadays.
> 4. The page isn't using the <Container> component.

@oliviertassinari I tried to address all issues with - https://github.com/mui-org/material-ui/pull/24095/commits/1adf45b4f254762617a4e11ff71aced5bd7c9d71 I think it looks closer now to what you had in mind

oliviertassinari commented:
> @oliviertassinari I tried to address all issues with - 1adf45b I think it looks closer now to what you had in mind

Yes, it does :). However, notice that the Grid can't be used for top layout, at least with a non zero spacing. The negative spacing introduces a scrollbar

<img width="125" alt="Capture d’écran 2020-12-28 à 19 40 57" src="https://user-images.githubusercontent.com/3165635/103236270-ba789300-4944-11eb-91c9-77fbb58f6d1f.png">

The `Container` hedges this and avoid the text to be close to the edges with built-in padding.

<img width="289" alt="Capture d’écran 2020-12-28 à 19 43 58" src="https://user-images.githubusercontent.com/3165635/103236416-1f33ed80-4945-11eb-9838-7f45a019fbc7.png">

---

Let me know when you want me to review the responsive behavior. .

mnajdova commented:
>  However, notice that the Grid can't be used for top layout, at least with a non zero spacing. The negative spacing introduces a scrollbar

Ah, I see the issue, will fix it tomorrow. I think I will wrap up everything tomorrow and we can move to reviewing the responsiveness 👍

oliviertassinari commented on packages/material-ui/src/Typography/Typography.d.ts:
```diff
@@ -117,6 +119,10 @@ export interface TypographyTypeMap<P = {}, D extends React.ElementType = 'span'>
      * @default false
      */
     paragraph?: boolean;
+    /**
+     * The system prop that allows defining system overrides as well as additional CSS styles.
+     */
+    sx?: SxProps<Theme>;
```
@mnajdova Should we take this change back to the component in an isolate PR? It was forgotten to in the Avatar component. 

On a different note, would it be simpler to do:

```diff
-import { SxProps } from '@material-ui/system';
-import { Theme } from '@material-ui/core/styles';
+import { BoxProps } from '@material-ui/core/Box';

-sx?: SxProps<Theme>;
+sx?: BoxProps['sx'];
```

mnajdova replied:
Oops, yes, will take a look tomorrow, I believe it is not added on the Buttons too. I don’t have a preference on which type we should use, as long as we add it :))

oliviertassinari commented:
I have squashed the 100 commits into 1. I'm continuing the effort. The plan is to get the about page ready by next week. It would be good to have 2 pages done by a freelancer, so we can focus on solving our core business problems. Us implementing 1 or 2 pages is probably enough before we hit diminishing returns and a high opportunity cost.

mnajdova commented on docs/pages/branding/about.tsx:
```diff
@@ -0,0 +1,1049 @@
+import BrandingRoot from 'docs/src/modules/branding/BrandingRoot';
+import BrandingBeginToday from 'docs/src/modules/branding/BrandingBeginToday';
+
+export default function Pagee() {
```
```suggestion
export default function Page() {
```

mnajdova commented on docs/src/modules/branding/BrandingBeginToday.tsx:
```diff
@@ -0,0 +1,78 @@
+      <Box
+        component="img"
+        src="/static/branding/block5.png"
+        alt=""
```
```suggestion
        alt="Decorative blocks"
```

oliviertassinari replied:
I have actually removed the alt for accessibility reasons, see https://html.spec.whatwg.org/multipage/images.html#a-purely-decorative-image-that-doesn't-add-any-information. The imagine is decorative.

mnajdova replied:
Ah wasn't aware of it. Should we then add `aria-hidden` at least on these images?

mnajdova replied:
I think we should based on https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute

oliviertassinari replied:
I could find #18745 touching on the same topic. We went with alt="" instead of aria-hidden. No idea why. @mbrookes @eps1lon What should we do here?

eps1lon replied:
Decorative images should have `alt=""`. `alt=""` signals that the image is decorative. `aria-hidden` is for things you actually want to hide. Since the image is visible to sighted users, it should also be "visible" to assistive technology. Screen readers should announce it as a decorative image if asked to.

eps1lon replied:
I think the MDN article shouldn't mention that just like you wouldn't recommend `<img role="presentation" />`: `alt=""` is more established.

mnajdova replied:
Thanks for the explanation 👍

mnajdova commented on docs/src/modules/branding/BrandingBeginToday.tsx:
```diff
@@ -0,0 +1,78 @@
+        <Box
+          component="img"
+          src="/static/branding/block1-blue.svg"
+          alt=""
```
alt="Decorative blocks"

mnajdova commented on docs/src/modules/branding/BrandingFooter.tsx:
```diff
@@ -0,0 +1,294 @@
+                </Link>
+              </li>
+              <li>
+                {/* TODO: add link */}
```
```suggestion
```

mnajdova commented on docs/src/modules/branding/BrandingFooter.tsx:
```diff
@@ -0,0 +1,294 @@
+                </Link>
+              </li>
+              <li>
+                {/* TODO: add link */}
```
```suggestion
```

mnajdova commented on docs/src/modules/branding/BrandingFooter.tsx:
```diff
@@ -0,0 +1,294 @@
+            <Typography component="h3">{t1('Explore')}</Typography>
+            <ul>
+              <li>
+                {/* TODO: add link */}
```
```suggestion
```

mnajdova commented on docs/src/modules/branding/BrandingFooter.tsx:
```diff
@@ -0,0 +1,294 @@
+                </Link>
+              </li>
+              <li>
+                {/* TODO: add link */}
```
```suggestion
```

oliviertassinari commented on docs/src/modules/branding/BrandingFooter.tsx:
```diff
@@ -0,0 +1,294 @@
+          </Grid>
+          <Grid item xs={6} sm={3} md={4} lg={2} className="BrandingFooter-list">
+            <Typography component="h3">{t1('Library')}</Typography>
+            {/* TODO: Should se make sections of the Sidebar valid links to target these? */}
```
```suggestion
```
